internal/querycoordv2/meta/target_manager.go

```diff
@@ -528,3 +529,43 @@ func (mgr *TargetManager) IsNextTargetExist(collectionID int64) bool {
 
 	return len(newChannels) > 0
 }
+
+func (mgr *TargetManager) SaveCurrentTarget(catalog metastore.QueryCoordCatalog) {
+	mgr.rwMutex.Lock()
+	defer mgr.rwMutex.Unlock()
+	if mgr.current != nil {
+		for id, target := range mgr.current.collectionTargetMap {
```

Comment:
collection target Map may be very large which is not friendly for etcd IO.

Reply:
1. `saveTarget` and `recoverTarget` only happens when qc restarts, the trigger frequency is low.
2. the `pbTarget` only contains (channel Names, segment, cp), which won't be too larger. for example, a collection with 30w segment's target is about 3MB.
3.  use `zstd` to compress and decompress the `pbTarget`, for example, the 3MB target could be compressed to 600KB.